File: Absa Backup/absa-banking-app-android/libmonad/src/main/java/styleguide/utils/ViewUtils.java
/*
 *  Copyright (c) 2018 Absa Bank Limited, All Rights Reserved.
 *
 *  This code is confidential to Absa Bank Limited and shall not be disclosed
 *  outside the Bank without the prior written permission of the Absa Legal
 *
 *  In the event that such disclosure is permitted the code shall not be copied
 *  or distributed other than on a need-to-know basis and any recipients may be
 *  required to sign a confidentiality undertaking in favor of Absa Bank
 *  Limited
 *
 */

package styleguide.utils;

import android.content.Context;
import android.content.res.Resources;
import android.os.Build;
import android.view.View;
import android.widget.TextView;

public final class ViewUtils {

    public static float getDisplayDensity() {
        return Resources.getSystem().getDisplayMetrics().density;
    }

    public static int generateViewId() {
        return View.generateViewId();
    }

    public static void setTextAppearance(TextView view, Context context, int resourceId) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
            view.setTextAppearance(resourceId);
        } else {
            view.setTextAppearance(context, resourceId);
        }
    }

}

File: Absa Backup/absa-banking-app-android/app/src/main/java/com/barclays/absa/utils/ViewUtils.kt
/*
 * Copyright (c) 2021 Absa Bank Limited, All Rights Reserved.
 *
 * This code is confidential to Absa Bank Limited and shall not be disclosed
 * outside the Bank without the prior written permission of the Absa Legal
 *
 * In the event that such disclosure is permitted the code shall not be copied
 * or distributed other than on a need-to-know basis and any recipients may be
 * required to sign a confidentiality undertaking in favor of Absa Bank
 * Limited
 */

package com.barclays.absa.utils

import android.view.View
import android.widget.ScrollView
import androidx.core.widget.NestedScrollView

object ViewUtils {
    fun scrollToTopOfView(scrollView: View, targetView: View) {
        when (scrollView) {
            is NestedScrollView -> scrollView.post { scrollView.smoothScrollTo(0, targetView.y.toInt()) }
            is ScrollView -> scrollView.post { scrollView.smoothScrollTo(0, targetView.y.toInt()) }
        }
    }

    fun scrollToBottomOfView(scrollView: View, targetView: View) {
        when (scrollView) {
            is NestedScrollView -> scrollView.post { scrollView.smoothScrollTo(0, targetView.y.toInt() + targetView.height) }
            is ScrollView -> scrollView.post { scrollView.smoothScrollTo(0, targetView.y.toInt() + targetView.height) }
        }
    }
}